test/charms/main/lib/charm.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+import os
+import base64
+import pickle
+
+from juju.framework import (
+    EventBase,
+    Event,
+)
+
+from juju.charm import (
+    CharmBase,
+    CharmEvents,
+)
+
+import logging
+
+logger = logging.getLogger()
+
+
+class DBRelationJoinedEvent(EventBase):
+    pass
+
+
+class DBRelationChangedEvent(EventBase):
+    pass
+
+
+class DBRelationDepartedEvent(EventBase):
+    pass
+
+
+class DBRelationBrokenEvent(EventBase):
+    pass
+
+
+class MonRelationJoinedEvent(EventBase):
+    pass
+
+
+class MonRelationChangedEvent(EventBase):
+    pass
+
+
+class MonRelationDepartedEvent(EventBase):
+    pass
+
+
+class MonRelationBrokenEvent(EventBase):
+    pass
+
+
+class HARelationJoinedEvent(EventBase):
+    pass
+
+
+class HARelationChangedEvent(EventBase):
+    pass
+
+
+class HARelationDepartedEvent(EventBase):
+    pass
+
+
+class HARelationBrokenEvent(EventBase):
+    pass
```

Comment:
A lot of wasted spacing here. Can we compress this a bit so we can read and reason about more of it at once?

Reply:
This is actually following the style guidelines for Python.  All linters will reject the `class Foo: pass` style and no blank lines between unless explicitly exempted (which will also affect all other cases where multi-statement lines are used and blank lines missing between class / function definitions).  I'm not sure that we want to go down that road, but maybe there's a linter that we can use which would support exempting only a specific block of code.

We could also get around it with per-line comments, a la:

```python
class InstallEvent(EventBase): pass  # noqa: ignore=E302,E701
class StartEvent(EventBase): pass  # noqa: ignore=E302,E701
class StopEvent(EventBase): pass  # noqa: ignore=E302,E701
class ConfigChangedEvent(EventBase): pass  # noqa: ignore=E302,E701
class UpdateStatusEvent(EventBase): pass  # noqa: ignore=E302,E701
class UpgradeCharmEvent(EventBase): pass  # noqa: ignore=E302,E701
class PreSeriesUpgradeEvent(EventBase): pass  # noqa: ignore=E302,E701
class PostSeriesUpgradeEvent(EventBase): pass  # noqa: ignore=E302,E701
class LeaderElectedEvent(EventBase): pass  # noqa: ignore=E302,E701
class LeaderSettingsChangedEvent(EventBase): pass  # noqa: ignore=E302,E701
```

Reply:
Got rid of this because Cory's PR got merged.